src/main/java/uk/gov/companieshouse/api/accounts/validation/DebtorsValidator.java

```diff
@@ -1,53 +1,79 @@
+import uk.gov.companieshouse.api.accounts.service.response.ResponseObject;
 import uk.gov.companieshouse.api.accounts.transaction.Transaction;
 import uk.gov.companieshouse.api.model.company.CompanyProfileApi;
 
 @Component
-public class DebtorsValidator extends BaseValidator {
+public class DebtorsValidator extends BaseValidator implements CrossValidator<Debtors> {
 
     @Value("${invalid.note}")
     private String invalidNote;
 
     @Value("${inconsistent.data}")
     private String inconsistentData;
 
+    @Value("${current.balancesheet.not.equal}")
+    private String currentBalanceSheetNotEqual;
+
+    @Value("${previous.balancesheet.not.equal}")
+    private String previousBalanceSheetNotEqual;
+
     private static final String DEBTORS_PATH = "$.debtors";
     private static final String DEBTORS_PATH_PREVIOUS = DEBTORS_PATH + ".previous_period";
     private static final String CURRENT_TOTAL_PATH = DEBTORS_PATH + ".current_period.total";
     private static final String PREVIOUS_TOTAL_PATH = DEBTORS_PATH_PREVIOUS + ".total";
     private static final String PREVIOUS_TRADE_DEBTORS = DEBTORS_PATH_PREVIOUS + ".trade_debtors";
-    private static final String PREVIOUS_PREPAYMENTS = DEBTORS_PATH_PREVIOUS + ".prepayments_and_accrued_income";
+    private static final String PREVIOUS_PREPAYMENTS =
+        DEBTORS_PATH_PREVIOUS + ".prepayments_and_accrued_income";
     private static final String PREVIOUS_OTHER_DEBTORS = DEBTORS_PATH_PREVIOUS + ".other_debtors";
-    private static final String PREVIOUS_GREATER_THAN_ONE_YEAR = DEBTORS_PATH_PREVIOUS + ".greater_than_one_year";
+    private static final String PREVIOUS_GREATER_THAN_ONE_YEAR =
+        DEBTORS_PATH_PREVIOUS + ".greater_than_one_year";
 
     private CompanyService companyService;
+    private CurrentPeriodService currentPeriodService;
+    private PreviousPeriodService previousPeriodService;
 
     @Autowired
-    public DebtorsValidator(CompanyService companyService) {
+    public DebtorsValidator(CompanyService companyService,
+                            CurrentPeriodService currentPeriodService,
+                            PreviousPeriodService previousPeriodService) {
         this.companyService = companyService;
+        this.currentPeriodService = currentPeriodService;
+        this.previousPeriodService = previousPeriodService;
     }
 
-    public Errors validateDebtors(@Valid Debtors debtors,
-                                  Transaction transaction) throws DataException {
+    public Errors validateDebtors(@Valid Debtors debtors, Transaction transaction,
+                                  String companyAccountsId,
+                                  HttpServletRequest request) throws DataException {
 
         Errors errors = new Errors();
 
+        crossValidate(errors, request, companyAccountsId, debtors);
+        crossValidatePreviousPeriod(errors, request, companyAccountsId, debtors);
```

Comment:
Just wondering if we need this here`crossValidatePreviousPeriod(errors, request, companyAccountsId, debtors)` 

this validation already occurs inside `crossValidate()`

Reply:
duplicated method call removed in 0b7cfe8